### ISSUE TITLE
Long string changes color if split it and add "\" to the end 

### ISSUE THREAD

**Issue by Akopov4** [microsoft/pylance-release#5733]
<h3> Environment data </h3>

-   Language Server version: 2024.4.1
-   OS and version: linux x64
-   Python version (and distribution if applicable, e.g. Anaconda): python3.8 , pip23.2.1
-   python.analysis.indexing: true
-   python.analysis.typeCheckingMode: basic

<h3> Code Snippet </h3>

`'sdfsdfsddsssdadsadsadsadasdasdsadsadasdasdasdasdasdasdasdasdadasdasdasdasadasdasdasdasdasdasdasdsadasdas\
       dasdsdsaadsad'`

```python
XXX
```

<h3> Repro Steps </h3>

1. Create long string .e.g: 'sdfsdfsddsssdadsadsadsadasdasdsadsadasdasdasdasdasdasdasdasdadasdasdasdasadasdasdasdasdasdasdasdsadasdasdasdsdsaadsad'
2. Pay attention to the color of symbols in the string
3. Split it with "\":
'sdfsdfsddsssdadsadsadsadasdasdsadsadasdasdasdasdasdasdasdasdadasdasdasdasadasdasdasdasdasdasdas\
       dsadasdasdasdsdsaadsad'

<h3> Expected behavior </h3>

Color of symbols shouldn't change

<h3> Actual behavior </h3>

Color of symbols doesn't change
<h3>Note </h3>
"editor.wordWrap": "wordWrapColumn"    doesn't add "\" to the string when string reaches the limit of  "editor.wordWrapColumn"
<h3> Logs </h3>

[vscode_log.txt](https://github.com/microsoft/pylance-release/files/14894864/vscode_log.txt)

```
XXX
```

<h3> GIF animation </h3>

![Peek 2024-04-06 22-57](https://github.com/microsoft/pylance-release/assets/17915103/72fd98cd-ab94-4ed4-9e9c-fcfabbc29fca)

```
XXX
```

**Comment by debonte** [comment]
This is a TextMate grammar issue.

For what it's worth, it seems that this only repros for loose string literals. If they are assigned to an indentifier, they are colored correctly:

```python
'sdfsdfsddsssdadsadsadsadasdasdsadsadasdasdasdasdasdasdasdasdadasdasdasdasadasdasdasdasdasdasdasdsadasdas\
     dasdsdsaadsad'

foo = 'sdfsdfsddsssdadsadsadsadasdasdsadsadasdasdasdasdasdasdasdasdadasdasdasdasadasdasdasdasdasdasdasdsadasdas\
     dasdsdsaadsad'

print('sdfsdfsddsssdadsadsadsadasdasdsadsadasdasdasdasdasdasdasdasdadasdasdasdasadasdasdasdasdasdasdasdsadasdas\
     dasdsdsaadsad')
```

![image](https://github.com/microsoft/pylance-release/assets/12820898/6b4ef59f-5191-48bc-bd86-68c274c1fe19)

**Comment by Akopov4** [comment]
Is it possible to add backslash when line reaches limit of wordWrap?

**Comment by debonte** [comment]
> @debonte Is it possible to add backslash when line reaches limit of wordWrap?

Please open a separate issue for that. We'll move it to discussions as an enhancement request and see if gets up votes from other users.

**Comment by Akopov4** [comment]
@debonte ok. Do I need to close this issue?

**Comment by debonte** [comment]
> @debonte ok. Do I need to close this issue?

No, let's let this one open to track the textmate grammar issue.

**Comment by RedCMD** [comment]
should add `\\n?` to `string-line-continuation`

<img width="463" height="144" alt="Image" src="https://github.com/user-attachments/assets/113b0989-25b7-4cc5-aad3-756bba017018" />

https://github.com/MagicStack/MagicPython/blob/7d0f2b22a5ad8fccbd7341bc7b7a715169283044/grammars/MagicPython.tmLanguage#L1493

here's my VSCode extension to help with authoring TextMate grammars https://marketplace.visualstudio.com/items?itemName=RedCMD.tmlanguage-syntax-highlighter

**Comment by heejaechang** [comment]
Repro workspace for investigation:
https://github.com/heejaechang/pylance-repro/tree/main/pylance-issues/5733

**Comment by heejaechang** [comment]
This issue has been fixed in prerelease version 2026.3.100, which we've just released. You can find the changelog here: [CHANGELOG.md](https://github.com/microsoft/pylance-release/blob/main/CHANGELOG.md#20263100-24-August-2026-PreRelease)
